modin/experimental/cloud/cluster.py

```diff
@@ -192,6 +194,7 @@ def __enter__(self):
         self.spawn(wait=True)  # make sure cluster is ready
         self.connection.activate()
         self.old_backends = set_backends(self.target_engine, self.target_partition)
+        Engine.subscribe(_update_engine)
```

Comment:
Local engine should be updated before working in remote context (if don't do this, unexpected errors appear).

```shell
========= Remote Traceback (1) =========
Traceback (most recent call last):
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/rpyc/core/protocol.py", line 320, in _dispatch_request
    res = self._HANDLERS[handler](self, *args)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/rpyc/core/protocol.py", line 593, in _handle_call
    return obj(*args, **dict(kwargs))
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/pandas/dataframe.py", line 109, in __init__
    Engine.subscribe(_update_engine)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/config/pubsub.py", line 213, in subscribe
    callback(cls)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/pandas/__init__.py", line 113, in _update_engine
    initialize_ray()
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/modin/engines/ray/utils.py", line 119, in initialize_ray
    ray.init(
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/ray/_private/client_mode_hook.py", line 82, in wrapper
    return func(*args, **kwargs)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/ray/worker.py", line 929, in init
    _global_node = ray.node.Node(
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/ray/node.py", line 181, in __init__
    session_name = _get_with_retry(redis_client, "session_name")
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/ray/node.py", line 41, in _get_with_retry
    result = redis_client.get(key)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/redis/client.py", line 1606, in get
    return self.execute_command('GET', name)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/redis/client.py", line 898, in execute_command
    conn = self.connection or pool.get_connection(command_name, **options)
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/redis/connection.py", line 1192, in get_connection
    connection.connect()
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/redis/connection.py", line 567, in connect
    self.on_connect()
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/redis/connection.py", line 643, in on_connect
    auth_response = self.read_response()
  File "/home/ubuntu/miniconda/envs/modin/lib/python3.8/site-packages/redis/connection.py", line 756, in read_response
    raise response
redis.exceptions.ResponseError: WRONGPASS invalid username-password pair
```